### PR TITLE
community[minor]: Add `DuckDB` as a vectorstore

### PR DESCRIPTION
**Pull request by Hugoberry**
DuckDB has a cosine similarity function along list and array data types, which can be used as a vector store.
    - **Description:** The latest version of DuckDB features a cosine similarity function, which can be used with its support for list or array column types. This PR surfaces this functionality to langchain.
    - **Dependencies:** duckdb 0.10.0
    - **Twitter handle:** @igocrite

### CODE REVIEW

File: libs/community/langchain_community/vectorstores/duckdb.py
Comment (eyurtsev) [inline libs/community/langchain_community/vectorstores/duckdb.py:16]: Would you mind adding a security notice here. See other security notices in doc-strings throughout the code base.

If I'm not mistaken, `duckdb` needs to be configured correctly otherwise the retriever will allow people to read files from the file system through `duckdb`

File: libs/community/langchain_community/vectorstores/duckdb.py
Comment (eyurtsev) [inline libs/community/langchain_community/vectorstores/duckdb.py:168]: Important: The default instance needs to be configured properly for security

we should via:  autoload_known_extensions, autoinstall_known_extensions, disabled_filesystems

image

**Review by eyurtsev — COMMENTED** [review]
Hi! Thank you for the contribution! This looks great.

Before we can merge we need to create the `duckdb` instance with safe default configuration. Let us know if you need help with it.

The reference is here: https://duckdb.org/docs/configuration/overview#configuration-reference

i added a bit more information in the comment near the instance that's being created


Some minor comments about fixing type hint

File: libs/community/langchain_community/vectorstores/duckdb.py
Comment (Hugoberry) [reply libs/community/langchain_community/vectorstores/duckdb.py:185]: I ended up setting and recommending `enable_external_access` to `false`, which seemed to be the most restricted option. I didn't set it for an existing connection, as this might interfere with the behaviour of the external code. 

I also mentioned it in the `security note`. 

Thanks for this recommendation.

**Comment by eyurtsev** [comment]
PR looks great! I'll do another security review by eow and if all good will merge!

**Review by eyurtsev — APPROVED** [review]

**Comment by eyurtsev** [comment]
merge when unit tests pass

**Comment by eyurtsev** [comment]
@Hugoberry thank you sorry didn't realize I introduced linting errors

**Comment by Hugoberry** [comment]
Hey @eyurtsev, thanks for helping me throughout this PR. It's my first contribution to an OSS project. Sorry for committing excessively. Hopefully, I got it this time around.

And by the way, your remark on DuckDB security came in handy on one of my work-related projects. Not many developers talk about these issues in the data space.

**Pull request merged**
